source/scripts/main.js

```diff
@@ -0,0 +1,196 @@
+            dietaryRestriction.checked = true;
+        }
+    }
+
+    for (let i = 0; i < intolerancesContainerElements.length; i++) {
+        const intoleranceRestriction = intolerancesContainerElements[i];
+        // If our restriction is in the list, then check it on the page
+        if (getIntolerancesRestrictions && getIntolerancesRestrictions.includes(intoleranceRestriction.value)) {
+            intoleranceRestriction.checked = true;
+        }
+    }
+}
+
+function clearCheckBoxes() {
+    let checkboxes = document.querySelector("#recipe-page-container > recipe-page")
+        .shadowRoot.querySelectorAll("#ingredients-list > ul > ol > input");
+    checkboxes.forEach(e => e.checked = false);
+}
+
+function hideSettings() {
+    const settings = document.getElementById("settings-container");
+    settings.style.visibility = "hidden";
+    // settings.style.transform = "translate(-100%)";
+}
+
+function showHome() {
+    hideSettings();
+    hideCookbooks();
+    hideRecipeCards();
+    showCategoryCards();
+    hideRecipePage();
+    document.getElementById('search-query').value = ''; //clears search result
+    const search = document.getElementById("search");
+    search.style.visibility = "visible";
+}
+
+function hideHome() {
+    hideCategoryCards();
+    const search = document.getElementById("search");
+    search.style.visibility = "hidden";
+}
+
+function showCookbooks() {
+    hideSettings();
+    hideHome();
+    hideRecipeCards();
+    hideRecipePage();
+    const cookbook = document.getElementById("cookbook-container");
+    cookbook.style.visibility = "visible";
+}
+
+function hideCookbooks() {
+    const cookbook = document.getElementById("cookbook-container");
+    cookbook.style.visibility = "hidden";
+}
+
+function showRecipePage() {
+    const recipePage = document.getElementById("recipe-page-container");
+    recipePage.style.visibility = "visible";
+}
+
+function hideRecipePage() {
+    const recipePage = document.getElementById("recipe-page-container");
+    recipePage.style.visibility = "hidden";
+}
+
+function showRecipeCards() {
+    const recipeCards = document.getElementById("recipe-card-container");
+    recipeCards.style.visibility = "visible";
+}
+function hideRecipeCards() {
+    const recipeCards = document.getElementById("recipe-card-container");
+    recipeCards.style.visibility = "hidden";
+}
+
+function showCategoryCards() {
+    const categoryCards = document.getElementById("category-wrapper");
+    categoryCards.style.visibility = "visible";
+}
+
+function hideCategoryCards() {
+    const categoryCards = document.getElementById("category-wrapper");
+    categoryCards.style.visibility = "hidden";
+}
+
+function updateSettings() {
+    const dietaryRestrictionList = [];
+    // Get all the inputs under the div
+    const dietaryContainerElements = document.getElementById('dietary-container').elements;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
    const dietaryContainerElements = document.getElementById("dietary-container").elements;
```